internal/monitors/kubernetes/cluster/metrics/deployments.go

```diff
@@ -13,10 +15,10 @@ import (
 
 func datapointsForDeployment(dep *v1beta1.Deployment) []*datapoint.Datapoint {
 	dimensions := map[string]string{
-		"metric_source":        "kubernetes",
-		"kubernetes_namespace": dep.Namespace,
-		"uid":             string(dep.UID),
-		"kubernetes_name": dep.Name,
+		"metric_source":              "kubernetes",
+		"kubernetes_namespace":       dep.Namespace,
+		"kubernetes_deployment_uid":  string(dep.UID),
```

Comment:
Actually since we are assuming UID is unique across all object kinds, just make this `kubernetes_uid`.  The `kubernetes_pod_uid` dim will have to remain the same since we already have built-in content going for it.